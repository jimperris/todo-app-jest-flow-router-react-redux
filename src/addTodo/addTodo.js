//@flow
import type {Dispatch} from '../types/basic'
import {connect} from 'react-redux'
import * as actions from '../actions/actions'
import AddTodo from './addTodo.jsx'

const mapDispatchToProps = (dispatch: Dispatch) => ({
  addTodo: (name: string) => dispatch(actions.addTodo(name))
})
const Component = connect(null, mapDispatchToProps)(AddTodo)
export default Component
