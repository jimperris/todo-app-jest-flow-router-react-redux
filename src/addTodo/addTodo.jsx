//@flow
import React, { Component } from 'react'
import type { Dispatch } from '../types/basic'
import { connect } from 'react-redux'
import * as actions from '../actions/actions'

type Props = {
  addTodo: (title: string) => void
}
type State = {
  value: string
}
export class AddTodo extends Component<Props, State> {
  value: HTMLInputElement
  state = {
    value: 'name'
  }
  handleClick = () => {
    this.props.addTodo(this.state.value)
  }
  handleChange = ({target}: SyntheticInputEvent<>) => {
    this.setState({value: target.value})
  }
  render() {
    return (
      <div>
        <input autoFocus value={this.state.value} onChange={this.handleChange} />
        <button type="submit" onClick={this.handleClick}>Submit</button>
      </div>
    )
  }
}

const mapDispatchToProps = (dispatch: Dispatch) => ({
  addTodo: (name: string) => dispatch(actions.addTodo(name))
})
const AddTodoCont = connect(null, mapDispatchToProps)(AddTodo)
export default AddTodoCont
