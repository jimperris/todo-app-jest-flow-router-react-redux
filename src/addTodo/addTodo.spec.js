//@flow
import React from 'react'
import {AddTodo} from './addTodo.jsx'
import { shallow, mount } from 'enzyme'
describe('addTodo component', () => {
  const spy = jest.fn()
  const wrap = mount(
    <AddTodo addTodo={spy} />
  )
  it('should match snapshot', () => {
    expect(wrap).toMatchSnapshot()
  })
  it('should call spy', () => {
    const button = wrap.find('button')
    button.simulate('click')
    expect(spy).toHaveBeenCalled()
  })
  it('should change input text and state with it', () => {
    const input = wrap.find('input')

    input.simulate('change', {target: {value: 'namea'}})
    expect(wrap.state('value')).toBe('namea')
  })
})
