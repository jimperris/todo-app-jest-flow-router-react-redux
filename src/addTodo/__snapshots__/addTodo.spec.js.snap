// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addTodo component should match snapshot 1`] = `
<div>
  <input
    autoFocus={true}
    onChange={[Function]}
    value="name"
  />
  <button
    onClick={[Function]}
    type="submit"
  >
    Submit
  </button>
</div>
`;
